diff --git a/src/nvim/main.c b/src/nvim/main.c
index 5b5c8a2..6c4bbab 100644
--- a/src/nvim/main.c
+++ b/src/nvim/main.c
@@ -273,7 +273,11 @@ int main(int argc, char **argv)
     printf(_("%d files to edit\n"), GARGCOUNT);
 
   full_screen = true;
-  t_colors = 256;
+  if (strcmp(getenv("TERM"), "linux") == 0) {
+    t_colors = 8;
+  } else {
+    t_colors = 256;
+  }
   check_tty(&params);
 
   /*
